presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/MetadataResolver.java

```diff
@@ -87,4 +90,24 @@ default boolean isMaterializedView(QualifiedObjectName viewName)
     {
         return getMaterializedView(viewName).isPresent();
     }
+
+    /**
+     * Returns list of registered types.
+     */
+    List<Type> getTypes();
+
+    /**
+     * Returns list of registered parametric types.
+     */
+    List<ParametricType> getParametricTypes();
```

Comment:
Javadoc should probably specify whether these List return values are expected to be mutable or not.